src/wasm-binary.h

```diff
@@ -1549,15 +1548,13 @@ class WasmBinaryBuilder {
     auto num = getU32LEB();
     for (size_t i = 0; i < num; i++) {
       Memory::Segment curr;
-      curr.offset = getU32LEB();
+      auto offset = getU32LEB();
       auto size = getU32LEB();
-      auto buffer = (char*)malloc(size);
+      char buffer[size];
```

Comment:
Small stack allocations are fine, but this one is controlled by the input, which is a Bad Idea. Dynamic allocation is better in those cases.